### PR TITLE
tests: add tests for reliable service under packet loss

### PR DESCRIPTION
**Pull request by ainghazal**
# Checklist

* [x] I have read the contribution guidelines
* [x] Iff you changed code related to services, or inter-service communication, make sure you update the diagrams in `ARCHITECTURE.md`.
* [x] Reference issue for this pull request:

# Description

Exercise reliable service passing a vector of packet ID to simulate packet loss.

Additionally, this PR updates github actions version to the latest action release for checkout (v4) and setup-go (v5). Eventually, we'll currently replace the test setup with the one we're using to run tests on the internal packages. 

TODO: This PR needs to be rebased on top of #60

### CODE REVIEW

**Comment by ainghazal** [comment]
I think this is ready for review. the only file of interest after merging #60 should be https://github.com/ooni/minivpn/pull/63/files#diff-b9a3711895939a4f4dc918dccb1b8e3eda12d2fea35a79984d7a7a7aab12a9a3 (modulo any minor changes)

File: internal/vpntest/packetio.go
Comment (ainghazal) [inline internal/vpntest/packetio.go:117]:
```suggestion
	pw.packetPayloadSize = size
	pw.WriteSequence(seq)
```

File: internal/reliabletransport/reliable_loss_test.go
Comment (bassosimone) [inline internal/reliabletransport/reliable_loss_test.go:185]:
```suggestion
		// TODO(ainghazal): exclude the following tests if not `-short`?
```

File: internal/reliabletransport/reliable_reorder_test.go
Comment (bassosimone) [inline internal/reliabletransport/reliable_reorder_test.go:22]: something's wrong in this diagram

File: internal/vpntest/packetio.go
Comment (bassosimone) [inline internal/vpntest/packetio.go:206]:
```suggestion
	// TODO(ainghazal): move the comparison to witness, leave only wait here
```

File: internal/vpntest/packetio_test.go
Comment (bassosimone) [inline internal/vpntest/packetio_test.go:315]:
```suggestion
```

**Review by bassosimone — APPROVED** [review]
🎻 🎻 🎻 🎻 

LGTM with the provided comments applied

File: internal/reliabletransport/reliable_reorder_test.go
Comment (ainghazal) [reply internal/reliabletransport/reliable_reorder_test.go]: your dear tabs :P fixed

**Pull request merged**
